up-l1/mqtt.adoc

```diff
@@ -35,8 +35,85 @@ MQTT is an OASIS standard messaging protocol for the Internet of Things (IoT). I
 
 For more information, please refer to https://mqtt.org/
 
-
 == Specifications
 
+=== UAttribute Mapping
+
+MQTT 5 supports custom header fields, and we leverage this to map certain UAttribute values into the MQTT header.
+
+[cols="1,1"]
+|===
+| MQTT Header Property Code | Value (key:value)
+| UserProperty
+| ("upriority","{UAttribute.priority}")
+| CorrelationData
+| ("reqId", "{UAttribute.reqId}")
+| UserProperty
+| ("sink","{UAttribute.sink}") # Should this be serialized to MicroUri?
+|===
+
+The upriority field is needed as mqtt's QoS is very limited, so this will allow listeners to prioritize messages.
+
+The Correlation Data Property Code contains the request correlation id.
+
+The sink field is used to correctly map a message to the correct uE as the topic name may not contain all of the information.
+
+=== URI Mapping
+
+Depending on whether the UURI is local or remote, the MQTT Topic that the message will be published on will be different:
+
+* The remote UUri (with UAuthority): `upr/UUri.UAthority.name`
```

Comment:
I'm less familiar with what's considered 'typical' for MQTT5 identifiers that are published on / subscribed to.

Over in the Zenoh spec + implementation, we went with Micro Form, since it's [there's this bit in the spec](https://github.com/eclipse-uprotocol/up-spec/tree/main/up-l1#211-parameters):
![image](https://github.com/eclipse-uprotocol/up-spec/assets/11622119/2431b80e-4572-41a7-aa27-7b70c4ad70b4)
which to my mind makes Micro Form kind of the bar to clear.

If Resolved URIs are available (satisfies both Micro + Long Forms) that works too, but that way at a minimum only Micro Form is needed.

This does degrade the "reading the URI on the wire in human-readable format" experience though.

WDYT?

Reply:
I went back and forth on this a bit, and I can see both sides (using micro uri vs a string representation). One thing to consider is that mqtt topics require a string representation so it had initially made sense to use the Authority name and if that became too cumbersome switch it to Micro Uri. One thing that was mentioned to me is that UTransport can expect a resolved uri in a message to transport (am I remembering this correctly @stevenhartley?) and if that is the case I dont think there is a ton of value in serializing the topic name.

Reply:
Yeah. Zenoh also requires a string for their key expression (KeyExpr). They end up serializing and chopping up, then arranging in a string. [Here's a link](https://github.com/eclipse-uprotocol/up-client-zenoh-rust/blob/8855b9abd4bd27228d30d9061522194f330fa547/src/lib.rs#L89-L116) to where that's done for reference.

Reply:
Related to #68

Specifically, [my last comment](https://github.com/eclipse-uprotocol/up-spec/issues/68#issuecomment-1994379074), with the remaining question from that issue.

Reply:
All of these discussions are making me believe we need to bring back the short form URI. This was a string version of the URI using the numbers in lieu of names (long) and in lieu of in byte[] (micro). 

Short form URI then would be:

`up://192.168.1.100/2/1.1/8000`

In the that example, the UAuthortiy is an IP address, the uService ID is 2 and version 1.1 and the topic ID is 8000. If this is valuable and can be used for MQTT, I can add it back to the spec and the language libraries but it will miss the 1.5.7 deadline tomorrow. @devkelley @sophokles73 @PLeVasseur WDYT?

Reply:
I updated the spec so the sink uri be short form serialized in the header.